### DISCUSSION TITLE
Python Version Upgrade

### DISCUSSION THREAD

**Discussion by nromyn** [tprelog/iocage-homeassistant#63] [Q&A] [answered]
This is a stupid question, I'm sure. What's the best method to upgrade the version of python that is used? I've been looking at the bash scripts and the post_install.sh specifies python 3.10 and the post_upgrade seems to point that way too, but I'm still on 3.9 as far as I can tell.

**Comment by tprelog** [comment, marked as answer]
While I have bumped the Python version in the plugin scripts, that only applies to a new installation

Upgrading python for an existing plugin requires manual intervention. On average this will take at least 15-30 minutes to complete. However a few users have reported waiting over an hour for the Home Assistant web UI to become available again! The amount of time it takes will vary depending on the speed of your NAS and the number of integrations you have added to Home Assistant.

Remember to take a snapshot of your jail before attempting to upgrade python
- replace `JAIL_NAME` with the name of your Home Assistant jail
- replace `SNAPSHOT_NAME` with something meaningful to yourself

```bash
iocage snapshot JAIL_NAME -n SNAPSHOT_NAME
```

Install the new python version then reinstall the Home Assistant service (virtualenv)

```bash
iocage console JAIL_NAME
# Press 0 to exit the menu
pkg install python310 py310-sqlite3
sysrc homeassistant_python=/usr/local/bin/python3.10
service homeassistant reinstall homeassistant
```

If something goes wrong you can revert to the snapshot you took before

```bash
iocage stop JAIL_NAME
iocage rollback JAIL_NAME -n SNAPSHOT_NAME
```

**Comment by nromyn** [comment]
Thanks, I owe you a beer!
